Source/WebInspectorUI/UserInterface/Views/ColorSwatchDisplay.js

```diff
@@ -0,0 +1,59 @@
+WI.ColorSwatchDisplay = class ColorSwatchDisplay extends WI.Object
+{
+    constructor(delegate, colorPalette)
+    {
+        super();
+
+        this._delegate = delegate;
+
+        this._element = document.createElement("div");
+        this._element.classList.add("swatch-display");
+
+        let swatchesTitle = this._element.appendChild(document.createElement("h2"));
+        swatchesTitle.innerText = WI.UIString("Variables", "Variables", "Title of swatches section in Color Picker");
+
+        let swatchWrapper = this._element.appendChild(document.createElement("ul"));
+        this._sortedPalette = this._specializedSort(colorPalette);
+
+        for (let paletteObject of this._sortedPalette) {
+            let swatch = new WI.InlineSwatch(WI.InlineSwatch.Type.Color, paletteObject.color, {readOnly: true, swatchTooltip: paletteObject.colorSource});
+            swatch.element.addEventListener("click", (event) => {
+                this._delegate.colorSwatchDispalyColorSelected(this, paletteObject.color, paletteObject.colorSource);
+            });
+            swatchWrapper.appendChild(document.createElement("li").appendChild(swatch.element));
+        }
+    }
+
+    get delegate() {return this._delegate;}
+
+    get element() {return this._element;}
+
+    _specializedSort(palette)
+    {   
+        const rotation = 2;
+        const weights = [Math.E / 11.279, Math.E / 3.934, Math.E / 39.975];
+
+        function visualAppeal (cssProperty) {
+            let color = cssProperty.color;
+            let [h, s, l] = color.hsl;
+
+            let luminosity = Math.floor(color.rgb.reduce((accum, currVal, i) => accum + currVal * weights[i]));
+
+            let [modH, modLuminosity, modLight] = [h, luminosity, l].map((component) => Math.floor(component * rotation));
```

Comment:
ditto about abbreviated variable names

is `mod` supposed to represent "modified" or "modulo" or something else?  this is one of the reasons why we dont abbreviate :)